Lib/test/test_functools.py

```diff
@@ -2033,9 +2035,9 @@ def __setitem__(self, key, value):
                 self.data[key] = value
             def clear(self):
                 self.data.clear()
-        _orig_wkd = functools.WeakKeyDictionary
+        _orig_wkd = weakref.WeakKeyDictionary
```

Comment:
Use `try ... finally` or `test.support.swap_attr()` for guaranteeing that weakref.WeakKeyDictionary is restored if the following checks fail.